src/group_theory/exponent.lean

```diff
@@ -0,0 +1,167 @@
+/-
+Copyright (c) 2021 Julian Kuelshammer. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Julian Kuelshammer
+-/
+import group_theory.order_of_element
+import algebra.punit_instances
+import algebra.gcd_monoid.finset
+import ring_theory.int.basic
+
+/-!
+# Exponent of a group
+
+This file defines the exponent of a group, or more generally a monoid. For a group `G` it is defined
+to be the minimal `n≥1` such that `g ^ n = 1` for all `g ∈ G`. For a finite group `G` it is equal to
+the lowest common multiple of the order of all elements of the group `G`.
+
+## Main definitions
+
+* `monoid.exponent_exists` is a predicate on a monoid `G` saying that there is some positive `n`
+  such that `g ^ n = 1` for all `g ∈ G`.
+* `monoid.exponent` defines the exponent of a monoid `G` as the minimal positive `n` such that
+  `g ^ n = 1` for all `g ∈ G`, by convention it is `0` if no such `n` exists.
+* `add_monoid.exponent_exists` the additive version of `monoid.exponent_exists`.
```

Comment:
What is the convention on docs of `to_additive` things?

Reply:
I'm not quite sure, but I remember being told that additive versions should be in the documentation, but I might be wrong.

Reply:
raised this on zulip [here](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/to_additive.20documentation)